### ISSUE TITLE
Move casper contract to use custom units

### ISSUE THREAD

**Issue by jacqueswww** [ethereum/casper#138]
Hi There!

So we recently introduced _custom units_ into vyper with [VIP 752](https://github.com/ethereum/vyper/issues/752). Quick example of usage https://vyper.readthedocs.io/en/latest/types.html#custom-unit-types.

What I would like to probe - if there is perhaps reason we can remove some of the current static unit types from vyper completely.

```
m
currency_value
currency1_value
currency2_value
```

I feel like these are quite opaque and will cause trouble for future users of vyper. Additionally because we force a description field on custom units, future users will be able to understand the units much better.

What is everyone's thoughts on this?
If there is consensus in agreement with the above proposal, I will create a VIP to remove the not-obvious static units from vyper.

**Comment by djrtwo** [comment]
I support this.

That said, I haven't been able to figure out a good custom unit for casper's scaled deposits. I guess maybe just `scale_factor` -- `wei / scale_factor`. Don't want to hijack this thread, but open to ideas.

**Comment by djrtwo** [comment]
cc: @ChihChengLiang @vbuterin

**Comment by ChihChengLiang** [comment]
Propose `x` for the unit, and `scale_factor` for the name of the unit. Prefer a short unit so we can avoid having a lengthy type `(wei/ lengthy_unit)` everywhere in the codebase. And the choice of `x` is as if we are saying this is a multiple of something, like 4x is 4 times and 5 wei/x is 4x scaled down of 20 wei.
```
# specify units used in the contract.
units: {
    x: "scale_factor"
}
```

**Comment by djrtwo** [comment]
@ChihChengLiang As for the move for vyper to get rid of all built-in units and only have custom ones. Do you support this?

**Comment by ChihChengLiang** [comment]
ah thanks Danny. I support this too. all custom units and no built-ins.

**Comment by djrtwo** [comment]
@jacqueswww I'm waiting on a new release from vyper to integrate some of the recent syntax changes and features. I'll also switch to custom units whenever a release comes.

**Comment by jacqueswww** [comment]
After we move to-eth-tester I will tag a new release ;)

**Comment by djrtwo** [comment]
yay so excited!

**Comment by djrtwo** [comment]
This is unblocked. Need to decide on our units now..

**Comment by djrtwo** [comment]
closing in favor of #153
